src/analysis/retail/priest/shadow/modules/spells/Voidform.tsx

```diff
@@ -1,39 +1,178 @@
 import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
-import Events, { ApplyBuffEvent } from 'parser/core/Events';
+import Events, {
+  ApplyBuffEvent,
+  RemoveBuffEvent,
+  DamageEvent,
+  UpdateSpellUsableEvent,
+} from 'parser/core/Events';
 import Abilities from 'parser/core/modules/Abilities';
 import TALENTS from 'common/TALENTS/priest';
 import { SpellLink } from 'interface';
 import GradiatedPerformanceBar from 'interface/guide/components/GradiatedPerformanceBar';
 import { explanationAndDataSubsection } from 'interface/guide/components/ExplanationRow';
 import SpellUsable from 'parser/shared/modules/SpellUsable';
+import SPELL_CATEGORY from 'parser/core/SPELL_CATEGORY';
+
+const rate = 0.0001;
 
 class Voidform extends Analyzer {
   static dependencies = {
     abilities: Abilities,
     spellUsable: SpellUsable,
   };
-
   protected abilities!: Abilities;
   protected spellUsable!: SpellUsable;
 
-  mindblast = 0;
-  casts = 0;
+  casts = 0; //casts of voidform
+  mindblast = 0; //number of mindblasts gained by entering voidform
+
+  oncestart = 0;
+  maxCastsVB = 0; //Max casts of Voidbolt
+
+  castVB = 0; //casts of Voidbolt
+  miss = 0; //missed potential casts of Void Bolt
+
+  VB = [0]; //timestamps of voidbolt spellusable updates
 
   constructor(options: Options) {
     super(options);
 
+    this.active = this.selectedCombatant.hasTalent(TALENTS.VOID_ERUPTION_TALENT);
+
+    //I don't know how to add an event listener at the start of the fight,
+    this.addEventListener(Events.damage, this.onFightstart);
+
+    this.addEventListener(
+      Events.UpdateSpellUsable.by(SELECTED_PLAYER).spell(SPELLS.VOID_BOLT),
+      this.onVBUpdate,
+    );
+
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.VOID_BOLT), this.onVBCast);
+
     this.addEventListener(
       Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.VOIDFORM_BUFF),
       this.enterVoidform,
     );
+
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.VOIDFORM_BUFF),
+      this.leaveVoidform,
+    );
+
+    this.addEventListener(Events.fightend, this.onFightend);
+
+    (options.abilities as Abilities).add({
+      spell: SPELLS.VOID_BOLT.id,
+      category: SPELL_CATEGORY.ROTATIONAL,
+      cooldown: (haste: number) => 6 / (1 + haste),
+      gcd: {
+        base: 1500,
+      },
+      castEfficiency: {
+        suggestion: true,
+        recommendedEfficiency: 0.85,
+        maxCasts: () => this.maxCastsVB,
+      },
+    });
+  }
+
+  onFightstart(event: DamageEvent) {
+    if (this.oncestart === 0) {
+      const startevent = event;
+      startevent.timestamp = this.owner.fight.start_time;
+      //At the start of the fight, VB is not available.
+      //Set VB to be on cooldown, and slow the cooldown rate so it remains unusable.
+      this.spellUsable.applyCooldownRateChange(SPELLS.VOID_BOLT.id, rate);
+      this.spellUsable.beginCooldown(startevent, SPELLS.VOID_BOLT.id);
```

Comment:
not sure how i feel about this

Reply:
It isn't great, but there seems to be no other way to make a spell not usable.  Without doing this, voidbolt shows as available for durations where it is not.
Incorrect:
![VBwrong](https://user-images.githubusercontent.com/103148620/233713078-fa9d23e1-1b61-46a2-9aa5-555eaa98aa3e.PNG)

Correct:
![VBcorrect](https://user-images.githubusercontent.com/103148620/233713318-0b48e331-7c8b-4648-8c4c-7c33c64c4513.PNG)